docs/express-motivation.qmd

```diff
@@ -0,0 +1,343 @@
+---
+title: Motivation
+editor:
+  markdown:
+    wrap: sentence
+format:
+  html:
+    include-in-header: assets/head-content.html
+---
+
+So far, we've really only seen Shiny _Express_: a simple way to learn and create basic apps.
+As we'll discuss here, the same things that make Express simple also make it difficult to manage complexity when your app grows in size and sophistication.[^streamlit-comp]
+And, although we'll cover some techniques for managing complexity and adding sophistication to Express, when you find yourself reaching for these techniques, it might be time to switch to Shiny Core.
+Shiny Core is a slightly more involved way to write apps using the same [components](ui-components.qmd) and [reactivity](reactive-foundations.qmd) you've already learned.
+The main difference is that Shiny Core embraces concepts like [decoupling](#decoupling) and [functional programming](#functional) which enforces programming patterns that, although require more effort upfront, will make it easier to manage complexity as your app grows.
+
+If you're already convinced that Shiny Core is right for your app, the [next article](express-difference.qmd) outlines the syntax differences and provides a guide for transitioning from Express to Core.
+
+[^streamlit-comp]: [Streamlit](https://streamlit.io/) is similar to Express in that it's easy to get started with, but difficult to manage complexity. See [this article](https://www.shinyapps.io/articles/streamlit-vs-shiny/) for a comparison between Streamlit and Shiny.
```

Comment:
I think I might make this a callout pointing to another article. I don't think Streamlit users are going to be interested in Core at all, so by the time they get here they'll be on board with Shiny. Also Express way way better at managing complexity than Streamlit, so I probably wouldn't equate them. Express is a lot closer to Core on this metric than it is to Streamlit.